source/replication.txt

```diff
@@ -104,30 +104,43 @@ Automatic Failover
 ------------------
 
 When a primary does not communicate with the other members of the set
-for more than 10 seconds, an eligible secondary will hold an election
+for more than 10 seconds, an eligible secondary holds an election
 to elect itself the new primary. The first secondary to hold an
 election and receive a majority of the members' votes becomes primary.
 
 .. include:: /includes/fact-replica-set-protocolVersion1.rst
 
 .. include:: /images/replica-set-trigger-election.rst
 
-Although the timing varies, the failover process generally completes
-within a minute. For instance, it may take 10-30 seconds for the
-members of a :term:`replica set` to declare a :term:`primary`
-inaccessible (see :rsconf:`~settings.electionTimeoutMillis`). One of
-the remaining secondaries holds an :term:`election` to elect itself as
-a new primary. The election itself may take another 10-30 seconds.
+Lowering the :rsconf:`~settings.electionTimeoutMillis` 
+replication configuration option from the default ``10000`` (10 seconds)
+can result in faster automated failover at the cost of increased
+frequency of elections.
 
-.. versionchanged:: 3.2
+The replica set can continue to serve read queries 
+if such queries are configured to 
+:ref:`run on secondaries <replica-set-read-preference>` while the
+primary is offline. The replica set cannot process write operations 
+until the election completes successfully. 
 
-   Starting in MongoDB 3.2, with the :ref:`replication election
-   enhancements <3.2-rel-notes-rs-enhancements>`, MongoDB reduces
-   replica set failover time. See :ref:`replication election
-   enhancements <3.2-rel-notes-rs-enhancements>` for details.
+.. include:: /includes/fact-retryable-writes-failover-election.rst 
 
-See :ref:`replica-set-elections` and
-:ref:`replica-set-rollbacks` for more information.
+Since MongoDB 3.2, elections typically complete within two seconds
+in clusters running with the default 
+:rsconf:`protocolVersion: 1 <protocolVersion>` . The total 
+time in which a cluster may operate without a primary should not 
+typically exceed 12 seconds.
```

Comment:
"The total time in which a cluster may operate without a primary should not typically exceed 12 seconds."

This sentence nullifies the sentence before it and leaves me confused as to whether 2 or 12 seconds is typical. The paragraph that follows this in the rendered view, which comes from the include fact-election-latency.rst provides sufficient hinting that elections might take longer in some circumstances.

Reply:
I'm wondering if we should frame this in reverse.

e.g., "The total time in which a cluster may operate without a primary should not typically exceed 15 seconds, and may resolve as quickly as within two seconds."

In a sense that has the same issue - I do think there is some value in setting a 'range' that users can generally expect. Alternatively, we can just stick to the 15-second upper-limit and let users be pleasantly surprised whenever its faster. @stbrody thoughts on the latter?

Reply:
One primary motivation for this ticket is that customers are raising concerns about the amount of time failover takes as reported by our docs and that this can cause friction in the sales process. I think we should report a middle number. The median is reasonable. In the ticket description Cailin suggests this is 10.5 - 11.5 seconds (with default settings) and only 5.5-6.5 seconds in Atlas.